src/audio/module_adapter/module_adapter.c

```diff
@@ -849,6 +916,25 @@ int module_adapter_trigger(struct comp_dev *dev, int cmd)
 		return PPL_STATUS_PATH_STOP;
 	}
 
+	if (mod->num_input_buffers > 1) {
+		bool sources_active;
+
+		sources_active = module_source_status_count(dev, COMP_STATE_ACTIVE) ||
+				 module_source_status_count(dev, COMP_STATE_PAUSED);
```

Comment:
if you only need `module_source_status_count()` to identify whether there are *any* sources with the required state, you don't need to count them, you can return from there as soon as you find the first match. But since we aren't likely to get more than two sources too often, this isn't a huge difference